src/renderers/dom/client/ReactMount.js

```diff
@@ -164,10 +175,11 @@ function setID(node, id) {
  * @internal
  */
 function getNode(id) {
-  if (!nodeCache.hasOwnProperty(id) || !isValid(nodeCache[id], id)) {
-    nodeCache[id] = ReactMount.findReactNodeByID(id);
+  if (nodeCache.hasOwnProperty(id)) {
```

Comment:
Maybe:
var node = getNodeIfCached(id);
if(node) return node;
else return nodeCache[id] = ReactMount.findReactNodeByID(id);

At the very least that saves an object lookup (object lookups are surprisingly expensive in javascript).  It also reads a little better IMO.